packages/frontend/src/state/wallet/updater.ts

```diff
@@ -0,0 +1,160 @@
+import Onboard from 'bnc-onboard'
+import Notify from 'bnc-notify'
+import BigNumber from 'bignumber.js'
+import { useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+import { ethers } from 'ethers'
+import Web3 from 'web3'
+import { useAppSelector } from '@hooks/useAppSelector'
+import { setWeb3Settings, setAddress } from './actions'
+import { Networks } from '../../types'
+import useInterval from '@hooks/useInterval'
+
+export default function Updater(): null {
+  const dispatch = useDispatch()
+  const networkId = useAppSelector(({ wallet }) => wallet.networkId)
+  const _onboard = useAppSelector(({ wallet }) => wallet.onboard)
+  const web3 = useAppSelector(({ wallet }) => wallet.web3)
+  const address = useAppSelector(({ wallet }) => wallet.address)
+
+  const onNetworkChange = (updateNetwork: number) => {
+    if (updateNetwork in Networks) {
+      dispatch(setWeb3Settings({ networkId: updateNetwork as Networks }))
+      if (_onboard) {
+        const network = updateNetwork === 1337 ? 31337 : updateNetwork
+        localStorage.setItem('networkId', network.toString())
+        _onboard.config({
+          networkId: network,
+        })
+      }
+    } else {
+      if (address === null || !_onboard) return
+      _onboard.walletCheck()
+      console.log('Unsupported network')
+    }
+  }
+
+  const onWalletUpdate = (wallet: any) => {
+    if (wallet.provider) {
+      window.localStorage.setItem('selectedWallet', wallet.name)
+      const provider = new ethers.providers.Web3Provider(wallet.provider)
+      provider.pollingInterval = 30000
```

Comment:
you could make the 30000 a constant variable

Reply:
Added constant for polling interval and pushed new commit